src/PdoDeleteResourceModel.php

```diff
@@ -0,0 +1,211 @@
+{
+    /*
+     * Provides PDO SQL DELETE functionality.
+     *
+     * @since [*next-version*]
+     */
+    use DeleteCapablePdoTrait;
+
+    /*
+     * Provides SQL DELETE query building functionality.
+     *
+     * @since [*next-version*]
+     */
+    use BuildDeleteSqlCapableTrait;
+
+    /*
+     * Provides SQL WHERE clause building functionality.
+     *
+     * @since [*next-version*]
+     */
+    use BuildSqlWhereClauseCapableTrait;
+
+    /*
+     * Provides PDO expression value hash map generation functionality.
+     *
+     * @since [*next-version*]
+     */
+    use GetPdoExpressionHashMapCapableTrait;
+
+    /*
+     * Provides PDO value hash string generation functionality.
+     *
+     * @since [*next-version*]
+     */
+    use GetPdoValueHashStringCapableTrait;
+
+    /*
+     * Provides SQL reference escaping functionality.
+     *
+     * @since [*next-version*]
+     */
+    use EscapeSqlReferencesCapableTrait;
+
+    /*
+     * Provides SqL condition rendering functionality (via a template).
+     *
+     * @since [*next-version*]
+     */
+    use RenderSqlExpressionCapableTrait;
+
+    /*
+     * Provides SQL table name storage functionality.
+     *
+     * @since [*next-version*]
+     */
+    use SqlTableAwareTrait;
+
+    /*
+     * Provides SQL field names storage functionality.
+     *
+     * @since [*next-version*]
+     */
+    use SqlFieldNamesAwareTrait;
+
+    /*
+     * Provides SQL field-to-column map storage functionality.
+     *
+     * @since [*next-version*]
+     */
+    use SqlFieldColumnMapAwareTrait;
+
+    /*
+     * Provides SQL expression template storage functionality.
+     *
+     * @since [*next-version*]
+     */
+    use SqlExpressionTemplateAwareTrait;
+
+    /*
+     * Provides string normalization functionality.
+     *
+     * @since [*next-version*]
+     */
+    use NormalizeStringCapableTrait;
+
+    /*
+     * Provides array normalization functionality.
+     *
+     * @since [*next-version*]
+     */
+    use NormalizeArrayCapableTrait;
+
+    /*
+     * Provides functionality for creating invalid argument exceptions.
+     *
+     * @since [*next-version*]
+     */
+    use CreateInvalidArgumentExceptionCapableTrait;
+
+    /*
+     * Provides string translating functionality.
+     *
+     * @since [*next-version*]
+     */
+    use StringTranslatingTrait;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param PDO                   $pdo                The PDO instance to use to prepare and execute queries.
+     * @param TemplateInterface     $expressionTemplate The template for rendering SQL expressions.
+     * @param string|Stringable     $table              The name of the table from which records will be deleted.
+     * @param string[]|Stringable[] $fieldColumnMap     A map of field names to table column names.
+     */
+    public function __construct(PDO $pdo, TemplateInterface $expressionTemplate, $table, $fieldColumnMap)
```

Comment:
Any missing throws, such as for invalid table or map?

Reply:
Yep, exactly the ones you mentioned. 👍